### PR TITLE
fix race conditions in tests that uses the same identifiers

### PR DESCRIPTION
**Pull request by haakonvt**
… for the  (win+linux test runners)

## Checklist:
- [x] Tests added/updated.

### CODE REVIEW

**Comment by haakonvt** [comment]
These keep failing, e.g. a day ago: https://github.com/cognitedata/cognite-sdk-python/actions/runs/7782388421/job/21218699237

File: tests/tests_integration/test_api/test_data_modeling/test_containers.py [hunk 1]
```diff
@@ -56,7 +57,7 @@ def test_apply_retrieve_and_delete(self, cognite_client: CogniteClient, integrat
         # Arrange
         new_container = ContainerApply(
             space=integration_test_space.space,
-            external_id="IntegrationTestContainer",
+            external_id="IntegrationTestContainer" + random_string(5),
```
Comment (erlendvollset) [inline tests/tests_integration/test_api/test_data_modeling/test_containers.py:60]: We don't want to create a new container for each test, this delete + recreate containers pattern that really only happens in tests has a bunch of undesireable effects on the system (e.g. elasticsearch mapping explosion). Let's keep it as a fixed externalId and remove the cleanup below.

**Review by erlendvollset — CHANGES_REQUESTED** [review]

File: tests/tests_integration/test_api/test_data_modeling/test_containers.py (hunk 1 above)
Comment (haakonvt) [reply tests/tests_integration/test_api/test_data_modeling/test_containers.py]: Updated now! Not much was left of the test after, so please have another look. There's also two other tests that seem to create+delete containers: `test_apply_failed_and_successful_task` and `test_load_and_create_only_required_fields`. I've left them as is - but let me know if you want them changed/removed as well 👌

Comment (haakonvt) [reply tests/tests_integration/test_api/test_data_modeling/test_containers.py]: ...I also... did some... comment deletion 👀 Don't tell Anders 🤭

Comment (erlendvollset) [reply tests/tests_integration/test_api/test_data_modeling/test_containers.py]: Yeah, let's update those aswell 👍

Comment (haakonvt) [reply tests/tests_integration/test_api/test_data_modeling/test_containers.py]: Done, without the create+delete, it was already covered by other tests, so I deleted both.

**Review by erlendvollset — APPROVED** [review]
Lgtm!

**Pull request merged**